### PR TITLE
migrate to VM datasource for docker-compose files

### PR DESCRIPTION
**Pull request by denisgolius**
#3920

### CODE REVIEW

File: dashboards/victoriametrics-cluster.json
Comment (hagen1778) [inline dashboards/victoriametrics-cluster.json:1]: There is no need to modify the dashboards. We still want them to be compatible with Prometheus datasource.
I think, we can do the datasource type change ad-hoc for docker compose when initing Grafana container via `entrypoint` directive.

File: deployment/docker/Makefile [hunk 1]
```diff
@@ -173,15 +176,22 @@ package-via-docker-386:
 remove-docker-images:
 	docker image ls --format '{{.Repository}}\t{{.ID}}' | awk '{print $$2}' | xargs docker image rm -f
 
-
 docker-single-up:
-	docker-compose -f deployment/docker/docker-compose.yml up
+	rm -rf deployment/docker/grafana-plugins/victoriametrics-datasource
+	wget -nc https://github.com/VictoriaMetrics/grafana-datasource/releases/download/v$(VM_DATASOURCE_VERSION)/victoriametrics-datasource-v$(VM_DATASOURCE_VERSION).tar.gz \
+	-O deployment/docker/grafana-plugins/victoriametrics-datasource-v$(VM_DATASOURCE_VERSION).tar.gz
+	tar -xzvf deployment/docker/grafana-plugins/victoriametrics-datasource-v$(VM_DATASOURCE_VERSION).tar.gz -C deployment/docker/grafana-plugins/
```
Comment (hagen1778) [inline deployment/docker/Makefile:183]: there are following issues with this change:
* it always re-downloads the plugin even if it already exists locally
* it repeats itself in the `docker-cluster-up` command

I suggest to move the download process to a separate Makefile command and do the downloading only if datasource doesn't exist already. Would be great to have an ability to automatically re-download dataource if version `VM_DATASOURCE_VERSION ` doesn't match existing files.

**Review by hagen1778 — CHANGES_REQUESTED** [review]
Please see review comments.

The change should also include README changes, to highlight that this env uses VictoriaMetrics datasource now. 
And also changes to CHANGELOG.

**Comment by hagen1778** [comment]
Any updates on this @denisgolius ?

File: deployment/docker/Makefile (hunk 1 above)
Comment (denisgolius) [reply deployment/docker/Makefile]: fixed

File: dashboards/victoriametrics-cluster.json
Comment (denisgolius) [reply dashboards/victoriametrics-cluster.json:1]: fixed

**Comment by hagen1778** [comment]
@denisgolius would you mind rebase your branch on master?

**Comment by denisgolius** [comment]
Closing this PR to avoid merge conflicts. Here is a new one https://github.com/VictoriaMetrics/VictoriaMetrics/pull/4651

**Pull request closed without merging**